Classes/Controller/AbstractController.php

```diff
@@ -234,11 +234,11 @@ protected function getMultipleImages()
      */
     protected function getCollection()
     {
-        /* @var $resourceFactory ResourceFactory */
-        $resourceFactory = $this->objectManager->get(ResourceFactory::class);
+        /* @var $fileCollectionRepository FileCollectionRepository */
+        $fileCollectionRepository = GeneralUtility::makeInstance(FileCollectionRepository::class);
 
         /* @var $collection \TYPO3\CMS\Core\Resource\Collection\AbstractFileCollection */
-        $collection = $resourceFactory->getCollectionObject((int) $this->cObjData['tx_generic_gallery_collection']);
+        $collection = $fileCollectionRepository->findByUid((int) $this->cObjData['tx_generic_gallery_collection']);
```

Comment:
It's not  `$this->cObjData['tx_generic_gallery_collection']` missing, but the resourceFactory failing to find anything with the given ID. 

```php
PHP Warning: Undefined array key 3 in [...]\web\typo3\sysext\core\Classes\Resource\ResourceFactory.php line 133
at TYPO3\CMS\Core\Resource\ResourceFactory->getCollectionObject(3)
in [...]\web\typo3conf\ext\generic_gallery\Classes\Controller\AbstractController.php line 241
```
(array key 3 is just the ID of the collection in that case)

It fails on line 133 in ResourceFactory. I don't exactly know why, but ResourceFactory doesn't find the collection, while FileCollectionRepository can find it without issues.  `$this->collectionInstances` in the ResourceFactory stays empty.

Reply:
I see. Thanks for the explanation. I will test this locally and merge it asap. This is an issue with PHP 8.0 or with 8.1 only?

Reply:
You're welcome! It is definitly a issue with PHP 8.0. Not so sure about 8.1 though, because I don't have a Typo3 Instance running with 8.1.  
Guessing on the fact, that 8.1 only adds minor changes to 8.0, it probably was an issue there as well. But I'd take that with a grain of salt without firm proof.

Reply:
See issue is that the TYPO3 core is STILL not really PHP 8 ready. It's a mess. 

As soon as ` if (!$this->collectionInstances[$uid]) {` is changed to `if (!isset($this->collectionInstances[$uid])) {` it will work. 

I've added an forge ticket: https://forge.typo3.org/issues/98603